tla/consistency/SingleNode.tla

```diff
@@ -0,0 +1,122 @@
+---- MODULE SingleNode ----
+\* A lightweight specification to define the externally visible behaviour of CCF
+\* This specification has been inspired by https://github.com/tlaplus/azure-cosmos-tla
+\* Where possible, naming should be consistent with https://microsoft.github.io/CCF/main/index.html
+\* SingleNode considers a single node CCF service, so no view changes
+
+EXTENDS ExternalHistory
+
+\* Upper bound of the number of client events
+\* Note that this abstract specification does not model CCF nodes so there's no
+\* constant for the number of nodes
+CONSTANT HistoryLimit
+
+\* Abstract ledgers that contains only client transactions (no signatures)
```

Comment:
Can we call those LedgerViews, or LedgerBranches?

Reply:
done in https://github.com/microsoft/CCF/pull/5699/commits/19d1a33d16fd14ec975b7a4585f72d9efdff0cfb